File: backend/app/schemas/inventory.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict


class InventoryItemOut(BaseModel):
    product_id: int
    on_hand: int
    allocated: int
    available: int

    model_config = ConfigDict(from_attributes=True)


class InventoryAdjustmentCreate(BaseModel):
    product_id: int
    quantity_delta: int
    reference: Optional[str] = None


class InventoryTransactionOut(BaseModel):
    id: int
    product_id: int
    quantity_delta: int
    reference: Optional[str]
    type: str
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class InventoryAdjustmentOut(BaseModel):
    item: InventoryItemOut
    transaction: InventoryTransactionOut

File: frontend/types/inventory.ts
export interface InventoryItem {
  product_id: number;
  on_hand: number;
  allocated: number;
  available: number;
}

export interface InventoryTransaction {
  id: number;
  product_id: number;
  quantity_delta: number;
  reference: string | null;
  type: string;
  created_at: string; // ISO datetime string
}

export interface InventoryAdjustmentCreate {
  product_id: number;
  quantity_delta: number;
  reference?: string | null;
}

export interface InventoryAdjustmentOut {
  item: InventoryItem;
  transaction: InventoryTransaction;
}

